src/async_impl/client.rs

```diff
@@ -1023,7 +1023,7 @@ impl Client {
     ///
     /// # Errors
     ///
-    /// This method fails whenever supplied `Url` cannot be parsed.
+    /// This method fails whenever the supplied `Url` cannot be parsed.
```

Comment:
I also kind of want to change `Url` to `url` or something because, if someone passes an honest-to-goodness `Url`, I'm guess it can't _not_ be parsed. The issue is if they pass a non-`Url` `url` I assume? But that's pretty pedantic (and also maybe wrong).